### PR TITLE
Feature/generate from csv

### LINKED ISSUE #75: update generate_from_csv()
enable for more than one gc's

### PR DESCRIPTION
**Pull request by Piranias**
Fix #75

**Changes proposed in this pull request**:
- update generate_from_csv.py
- add option to assign vehicle_id if not given
- add options to assign consumption via soc, delta_soc or mileage and distance
- add option to export csv with vehicle_id
- update rtd
- update tests

The following steps were realized, as well (required):
- [x] Correct linting with `fake8 path_to_script`
- [x] Check if tests pass locally (`pytest ./src/tests.py`)

Also the following steps were realized (if applies):
- [x] Write docstrings to your code
- [x] For new functionalities: Explain in readthedocs
- [x] Write test(s) for your new patch of code

*Please mark above checkboxes as following:*
- [ ] Open
- [x] Done

### CODE REVIEW

**Review by jakobgemassmer — CHANGES_REQUESTED** [review]

File: generate_from_csv.py [hunk 1]
```diff
@@ -51,91 +71,110 @@ def generate_from_csv(args):
...
+            "estimated_time_of_departure": None,
+            "desired_soc": args.min_soc,
+            "soc": args.min_soc,
+            "vehicle_type": vt
+        }
 
+        cs_power = max([v[1] for v in vehicle_types[vt]['charging_curve']])
+        charging_stations[cs_name] = {
+            "max_power": cs_power,
+            "min_power": 0.1 * cs_power,
```
[the remark this answers is not in the document]
Comment (Piranias) [reply generate_from_csv.py]: I added a parameter to the argument parser / cfg file cs_power_min

File: generate_from_csv.py [hunk 2]
```diff
@@ -11,7 +11,21 @@
...
+    note: only one grid connector supported. Each line in the csv resembles one trip. The vehicle is
+    assigned by the vehicle_id. If the column vehicle_id is not given, the trips are assigned to the
+    vehicles by the principle: first in, first out. Note that in this case a minimum standing time
+    can be assigned to control the minimum time a vehicle can charge at the depot.
+
+    Needed columns:
+    - departure time (datetime)
+    - arrival time (datetime)
+    - vehicle_type (as in vehicle_types.json
+    - soc or delta_soc (optional, if not given the milage is taken instead) in [0,1]
```
[the remark this answers is not in the document]
Comment (Piranias) [reply generate_from_csv.py]: it's the soc at arrival. I added an explanation

File: generate_from_csv.py [hunk 3]
```diff
@@ -356,6 +444,10 @@ def csv_to_dict(csv_path):
     parser.add_argument('--battery', '-b', default=[], nargs=2, type=float, action='append',
                         help='add battery with specified capacity in kWh and C-rate \
                         (-1 for variable capacity, second argument is fixed power))')
+    parser.add_argument('--gc-power', type=int, default=530, help='set power at grid connection '
+                                                                  'point in kW')
+    parser.add_argument('--cs-power-min', type=int, default=0, help='set minimal power at charging '
```
Comment (stefansc1) [inline generate_from_csv.py:449]: Might make that of type float

File: generate_from_csv.py [hunk 4]
```diff
@@ -374,6 +466,10 @@ def csv_to_dict(csv_path):
     parser.add_argument('--include-price-csv-option', '-po', metavar=('KEY', 'VALUE'),
                         nargs=2, default=[], action='append',
                         help='append additional argument to price signals')
+    parser.add_argument('--min-standing-time', type=int, default=None,
```
Comment (stefansc1) [inline generate_from_csv.py:469]: Might also be float instead of integer?

**Review by stefansc1 — CHANGES_REQUESTED** [review]

**Pull request merged**